### PR TITLE
refactor(core,schemas): update roles table schemas, add type column

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

update roles table schemas, add type col.
Also updated UTs as well since `role` should come with `type` after this change.

## Testing

Tested locally, both seed DB and alteration from old DB status.

## Checklist

- [x] `.changeset`
- [x] unit tests
- [ ] integration tests
- [ ] docs

OR

- [ ] This PR is not applicable for the checklist

### CODE REVIEW

**Comment by charIeszhao** [comment]
Add a `changeset` maybe?

File: packages/schemas/alterations/next-1692351009-add-type-col-to-roles-table.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+enum InternalRole {
+  Admin = '#internal:admin',
+}
+
+const alteration: AlterationScript = {
+  up: async (pool) => {
+    // Get all m2m role ids.
+    const { rows: assignedM2mRoleIds } = await pool.query<{ id: string }>(sql`
+      select roles.id as "id" from roles
+        join applications_roles on applications_roles.role_id = roles.id and applications_roles.tenant_id = roles.tenant_id
```
Comment (wangsijie) [inline packages/schemas/alterations/next-1692351009-add-type-col-to-roles-table.ts:15]: left join?

**Review by wangsijie — APPROVED** [review]

File: .changeset/violet-mugs-tickle.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@logto/schemas": minor
+---
+
+Add `type` field to `roles` schema.
```
Comment (gao-sun) [inline .changeset/violet-mugs-tickle.md:5]: I think changeset should be some specific fix, or briefly describe a feature. This is too simplified.

File: .changeset/violet-mugs-tickle.md
Comment (gao-sun) [inline .changeset/violet-mugs-tickle.md:1]: Also the PR title can use "column" instead of "col"

**Review by gao-sun — APPROVED** [review]
changeset lgtm

File: packages/core/src/queries/applications-roles.ts [hunk 3]
```diff
@@ -20,18 +20,26 @@ export const createApplicationsRolesQueries = (pool: CommonQueryMethods) => {
...
+          ),
+          sql`, `
+        )}
+      `);
+    } catch (error: unknown) {
+      if (error instanceof CheckIntegrityConstraintViolationError) {
+        throw new InsertionError(ApplicationsRoles);
+      }
+      throw error;
+    }
```
Comment (simeng-li) [inline packages/core/src/queries/applications-roles.ts:41]: move to koaErrorHandler>

Comment (darcyYe) [reply packages/core/src/queries/applications-roles.ts]: Refactored

**Pull request merged**